### PR TITLE
Fix hidden size and support mistral nemo

### PR DESCRIPTION
**Pull request by AllentDan**
https://github.com/InternLM/lmdeploy/issues/2094

### CODE REVIEW

File: lmdeploy/turbomind/deploy/target_model/base.py [hunk 1]
```diff
@@ -195,9 +196,15 @@ def get_config(self, cfg: TurbomindModelConfig) -> TurbomindModelConfig:
             emb = bin.tok_embeddings()
             if emb is not None:
                 _vocab_size, dim = emb.shape
-                head_num = dim // cfg.size_per_head
+                # legacy, may get removed
+                head_num = final_cfg['head_num'] or dim // cfg.size_per_head
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/target_model/base.py:200]: I think we can remove "or dim // cfg.size_per_head"

File: lmdeploy/turbomind/deploy/source_model/deepseek_vl.py [hunk 2]
```diff
@@ -69,6 +70,7 @@ def model_info(self):
                     norm_eps=norm_eps,
                     attn_head_num=attn_head_num,
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/deepseek_vl.py:71]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/source_model/glm4.py [hunk 3]
```diff
@@ -100,6 +101,7 @@ def model_info(self):
                     norm_eps=norm_eps,
                     attn_head_num=attn_head_num,
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/glm4.py:102]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/source_model/internvl.py [hunk 4]
```diff
@@ -80,6 +81,7 @@ def model_info(self):
 
         return dict(num_layer=num_layer,
                     norm_eps=norm_eps,
+                    hidden_units=hidden_units,
                     attn_head_num=attn_head_num,
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/internvl.py:85]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/source_model/qwen.py [hunk 5]
```diff
@@ -77,6 +78,7 @@ def model_info(self):
             use_logn_attn = int(config['use_logn_attn'])
         return dict(num_layer=num_layer,
                     norm_eps=norm_eps,
+                    hidden_units=hidden_units,
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/qwen.py:81]: add `head_num=attn_head_num`

File: lmdeploy/turbomind/deploy/source_model/deepseek_vl.py (hunk 2 above)
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/deepseek_vl.py:71]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/source_model/glm4.py (hunk 3 above)
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/glm4.py:102]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/source_model/internvl.py (hunk 4 above)
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/internvl.py:85]: head_num=attn_head_num

File: lmdeploy/turbomind/deploy/target_model/base.py [hunk 6]
```diff
@@ -195,9 +196,15 @@ def get_config(self, cfg: TurbomindModelConfig) -> TurbomindModelConfig:
...
+                hidden_units = final_cfg[
+                    'hidden_units'] or head_num * cfg.size_per_head
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/target_model/base.py:202]: remove "or head_num * cfg.size_per_head"

File: lmdeploy/turbomind/deploy/target_model/base.py (hunk 1 above)
Comment (AllentDan) [reply lmdeploy/turbomind/deploy/target_model/base.py:200]: I don't know if it is ok to remove `dim // cfg.size_per_head`.

**Review by lvhan028 — APPROVED** [review]

**Review by lzhangzz — APPROVED** [review]

File: lmdeploy/turbomind/deploy/target_model/base.py (hunk 1 above)
Comment (lvhan028) [reply lmdeploy/turbomind/deploy/target_model/base.py]: It's fine, although I don't remember why I did that 😂

**Pull request merged**